### DISCUSSION TITLE
Be able to run `nx print-affected` with ` npx` for fast CI

### DISCUSSION THREAD

**Discussion by Lp-Francois** [nrwl/nx#23357] [Feature Requests]
## Description
In CI, we want to be able to determine if some apps were affected by the changes without having to install the node_modules which can be really big in the case of monorepo (in our case 3G)

```sh
$ npx nx print-affected --type=app --select=projects --plain --base $AFFECTED_BASE --head HEAD

 >  NX   Could not find Nx modules in this workspace.

   Have you run npm/yarn install?
```

## Motivation
Speed up the CI process and improve Developers experience. Especially with big monorepo, downloading the cached node_modules takes quite some time.

Especially with CI like vercel where it is possible to use a script checking if we should skip the build, running `print-affected fast enough is really helpful. 

Related issue: https://github.com/nrwl/nx/discussions/8448 where running npm install will install the whole node_modules, which we don't want.

### Current workaround we found
We moved the package.json and package-json.lock to a .bk extension, installed `nx` and then ran the command `print-affected` without error.

Related issues: from 2 years ago with the deprecated command - https://github.com/nrwl/nx/issues/5752

**Comment by Weetbix** [comment]
This is an important feature for monorepo tooling, our npm install takes upwards of 4 minutes before we can determine if a task even needs to be run.

**Comment by zwily** [comment]
@Lp-Francois I'm trying your workaround, but my `nx` command is failing without printing anything:

```npx --yes nx show projects --affected --base=HEAD^ --projects "apps/*"```

Any idea what's going on? Does nx need to be installed locally in the project to run an affected command? (`npx --yes nx` is installing a global one according to `npx --yes nx --version`)

Edit to add: This command works fine locally.

**Comment by Lp-Francois** [comment]
Hey, I am not sure why it doesn't work for you 🤔 
In local it works for me

Maybe try to add nx in your node_modules before running the workaround?

**Comment by martinmicunda** [comment]
The same feature was requested in this closed issue https://github.com/nrwl/nx/issues/4710 and https://github.com/nrwl/nx/issues/16381

**Comment by FrozenPandaz** [comment]
We need more discussion on this topic.

1. There is [nx-ignore](https://github.com/nrwl/nx-labs/tree/main/packages/nx-ignore) which handles this.
2. At this point, at least some packages will  have to be installed to properly create the graph.
3. It's easy for this to have false negatives.
